### PR TITLE
Doxygen duplicate tag strings issue

### PR DESCRIPTION
**Pull request by peter-michalski**
This is a duplicate of the previous Doxygen duplicate strings issue. I needed to move this work from my master into a dev branch. The original pull request was deleted during this move.

### CODE REVIEW

File: code/drasil-utils/lib/Utils/Drasil/Document.hs [hunk 1]
```diff
@@ -11,6 +11,14 @@ blank = text ""
 indent :: Doc -> Doc
 indent = nest 4
 
+-- | Helper which concatenates two Docs.
```
Comment (JacquesCarette) [inline code/drasil-utils/lib/Utils/Drasil/Document.hs:14]: Why not use `<>` directly then?

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
While this does satisfy the raw requirement of removing duplicate strings, it also doesn't really move the needle towards making this a generator move very much. But that can be done later.

File: code/drasil-utils/lib/Utils/Drasil/Document.hs (hunk 1 above)
Comment (peter-michalski) [reply code/drasil-utils/lib/Utils/Drasil/Document.hs:14]: <> could certainly be used directly 

`a +:+. b = a <> b <> text "."`

Sine I was creating a helper which concatenates two Docs and appends a period, I thought that a situation may arise where a helper for concatenating two Docs would also be needed. Similar to `foldlSent` in `Fold.hs`

Comment (peter-michalski) [reply code/drasil-utils/lib/Utils/Drasil/Document.hs]: Updated the PR, removing unnecessary functions.

Comment (JacquesCarette) [reply code/drasil-utils/lib/Utils/Drasil/Document.hs]: The point is that `Doc` already has a rich API, which we should use directly. We perhaps under-use it. `Sentence` is our own type, so we're forced to provide combinators for it ourselves.

File: code/drasil-utils/lib/Utils/Drasil/Document.hs [hunk 2]
```diff
@@ -11,14 +11,6 @@ blank = text ""
 indent :: Doc -> Doc
 indent = nest 4
 
--- | Helper which concatenates two Docs.
-(+:+) :: Doc -> Doc -> Doc
-a +:+ b = a <> b
-
--- | Helper which concatenates two Docs and appends a period.
-(+:+.) :: Doc -> Doc -> Doc
-a +:+. b = a +:+ b <> text "."
```
Comment (JacquesCarette) [inline code/drasil-utils/lib/Utils/Drasil/Document.hs]: Except that you went too far! I did want this particular combinator to be created.

File: code/drasil-code/lib/Language/Drasil/Code/Imperative/Doxygen/Import.hs [hunk 3]
```diff
@@ -33,8 +32,8 @@ verbosityToDoc Quiet = yes
 
 -- | Renderings of comments regarding the default tag values for various Doxygen settings.
 defNo, defYes :: Doc
-defNo = defaultValSentence +:+. no
-defYes = defaultValSentence +:+. yes
+defNo = defaultValSentence <> no <> text "."
```
Comment (JacquesCarette) [inline code/drasil-code/lib/Language/Drasil/Code/Imperative/Doxygen/Import.hs:35]: No, this particular change is a regression, it was better before.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: code/drasil-utils/lib/Utils/Drasil/Document.hs (hunk 2 above)
Comment (peter-michalski) [reply code/drasil-utils/lib/Utils/Drasil/Document.hs]: If I add the following combinator, then I will also need to add `+:+`, as I had in the previous PR.

```
-- | Helper which concatenates two Docs and appends a period.
(+:+.) :: Doc -> Doc -> Doc
a +:+. b = a +:+ b <> text "." 
```

Or would you prefer the following for now.
```
-- | Helper which concatenates two Docs and appends a period.
(+:+.) :: Doc -> Doc -> Doc
a +:+. b = a <> b <> text "." 
```

Comment (JacquesCarette) [reply code/drasil-utils/lib/Utils/Drasil/Document.hs]: I prefer the 2nd one.

Comment (peter-michalski) [reply code/drasil-utils/lib/Utils/Drasil/Document.hs]: The PR is now updated to reflect the requested change.

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**